src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/LockFileDependencyComparerWithoutContentHash.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Shared;
+
+namespace NuGet.ProjectModel.ProjectLockFile
+{
+    public class LockFileDependencyComparerWithoutContentHash : IEqualityComparer<LockFileDependency>
+    {
+        public static LockFileDependencyComparerWithoutContentHash Default { get; } = new LockFileDependencyComparerWithoutContentHash();
+
+        public bool Equals(LockFileDependency x, LockFileDependency y)
+        {
+            if (x == null || y == null)
+            {
+                return x == null && y == null;
+            }
+
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            return StringComparer.OrdinalIgnoreCase.Equals(x.Id, y.Id) &&
+                EqualityUtility.EqualsWithNullCheck(x.ResolvedVersion, y.ResolvedVersion) &&
+                EqualityUtility.EqualsWithNullCheck(x.RequestedVersion, y.RequestedVersion) &&
+                EqualityUtility.SequenceEqualWithNullCheck(x.Dependencies, y.Dependencies) &&
+                x.Type == y.Type;
+        }
+
+        public int GetHashCode(LockFileDependency obj)
```

Comment:
I think your intention was to generate a hash code for a `LockFileDependency` instance considering all properties except `ContentHash`.  Is that correct?  If so, remove this and add it to `LockFileDependency` in such a way that there are two methods:

    public override int GetHashCode();
    public int GetHashCodeWithoutContentHash();

...and they can call one shared implementation.

    private int GetHashCode(bool includeContentHash);

.  It does not belong here.  `LockFileDependency` already defines this!.